src/pages/_app.js

```diff
@@ -1,13 +1,56 @@
 import App, { Container } from 'next/app'
 import React from 'react'
 
+import Cookie from 'js-cookie'
+
+// Temporary fix to fetch dependency on 'fetches' lib
+if (typeof window === 'undefined') {
+  const fetch = require('isomorphic-fetch')
+}
+
+import { createClient, getHTTPMethods } from 'fetches'
+import { Provider } from 'react-fetches'
+
 class ProjectApp extends App {
+  static getInitialProps(context) {
+    const { ctx } = context
+    const getTokenFromLocalCookie = () => Cookie.get('token')
+    const getTokenFromServerCookie = req => {
+      if (!req.headers.cookie) {
+        return undefined
+      }
+      const sessionToken = req.headers.cookie.split(';').find(c => c.trim().startsWith('token='))
```

Comment:
Maybe we can set this word `token` as a env var, because we needed to hardcoded it in two different places, what can lead to future typos problems, what you think?